### PR TITLE
rpc-probe fixes and kvstore-fixtures and tests

### LINKED ISSUE #756: Make rpc-probe a first-class testing tool
Enhances #723 .

* Move `rpc-probe` under `tools/rpc-probe`.
* Make it create its files under `rpc/tests/kvstore-fixtures`.
* Create an `rpc-tests/kvstore.rs` test file that reads all fixtures and JSON deserializes them.
* ~~Make the binary emit a `version.json` file that has the Tendermint version it ran against.~~
* Possibly make rpc-probe set up docker with the kvstore endpoint (a la the `kvstore-test` crate using `cargo-make`).
* Fix all the folder traversals `/` -> `PathBuf::join` for correct cross-platform exexcution.

(Possibly in a new issue: repeat steps 2-5 and create a gaiad-fixtures using the gaiad endpoint.)

Note: I've crossed off the version control file. The reason: `rpc-probe` doesn't know exactly what endpoint it's running against. The docker image is defined in `cargo-make`. We'll have to rely on that for now.

### PR DESCRIPTION
**Pull request by greg-szabo**
Closes #756 .
Part of #723 .

* Important: all the JSON files in the change are auto-generated by `rpc-probe`. Do not review them in detail. ([Here's a link](https://github.com/informalsystems/tendermint-rs/pull/758/files?file-filters%5B%5D=.md&file-filters%5B%5D=.rs&file-filters%5B%5D=.sh&file-filters%5B%5D=.toml) where json files are filtered out.)

* Moved the `rpc-probe` folder into `tools/rpc-probe`.
* Removed the `run-with-docker.sh` script and implemented docker management using `cargo-make` in `tools/rpc-probe/Makefile.toml`.
* Changed `rpc-probe` to dump its results into `rpc/tests/kvstore_fixtures` by default.
* Implemented `rpc/tests/kvstore_fixtures.rs` tests that parse the rpc-probe results into Tendermint structs and make assertions against them.
* `rpc-probe` now generates requests with `null` input instead of `{}` where there is no parameter for the request. (Deserialization is implemented for `null`s.)
* `abci::transaction:Hash` now has the `PartialEq` derive for easier testing.
* `AbciQuery::prove` now has default serialization. (That was a small bug.)

* [X] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [X] Wrote tests
* [x] Updated CHANGELOG.md

### CODE REVIEW

**Review by thanethomson — COMMENTED** [review]
Looks good overall!

File: tools/rpc-probe/src/main.rs [hunk 1]
```diff
@@ -50,7 +50,18 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     };
     SimpleLogger::new().with_level(log_level).init().unwrap();
 
-    quick_probe_plan(&opts.output, Duration::from_millis(opts.request_wait))?
+    let output = if opts.output.to_str().unwrap() == "<rpc tests folder>" {
```
[the remark this answers is not in the document]
Comment (greg-szabo) [reply tools/rpc-probe/src/main.rs:53]: BTW, I would like to keep the defaults at the exact value we'd normally use, so the scripts around using `rpc-probe` can remain simple. Obviously, we could fall back to `probe-results` and then give the output at each run, but I think it's better devuser experience to have usable defaults.

I can revert back to `probe-results` and then set things up in `cargo-make`, if this is unappealing.

[the remark this answers is not in the document]
Comment (thanethomson) [reply tools/rpc-probe/src/main.rs:53]: You could create a type to represent the path, and then provide a `Default` implementation for it: https://docs.rs/structopt/0.3.21/structopt/#default-values

Then you can define the behaviour you want.

Not super necessary right now, perhaps I can add this in a follow-up PR.

[the remark this answers is not in the document]
Comment (greg-szabo) [reply tools/rpc-probe/src/main.rs]: Nah, I did it. Back when I was looking into how to do it, the idea of a new struct with a better default value came up but I dismissed it as too much work/code for very little benefit. But it's always a good idea to have proper code.

See the below commit, that should do it. Shall I move the struct to a separate file? (It's not used beyond `main.rs`.)

[the remark this answers is not in the document]
Comment (thanethomson) [reply tools/rpc-probe/src/main.rs]: Looks good! 👍

**Review by thanethomson — APPROVED** [review]

**Pull request merged**